Evaluate the channel name as a regex

oliwer opened the pull request:
This allows the user to write `somafm listen groove` instead of `somafm listen groovesalad`. It's very convenient. The regex forces the user to write the first letters of the channel, it cannot match
the middle or the end. This is voluntary and should help avoiding unexpected results.

oliwer committed:
Evaluate the channel name as a regex

This allows the user to write `somafm listen groove` instead of
`somafm listen groovesalad`. It's very convenient. The regex forces
the user to write the first letters of the channel, it cannot match
the middle or the end. This is voluntary and should help avoiding
unexpected results.

rockymadden commented:
@oliwer Love it. Are there any instances where a channel name is the prefix of another? I can't think of one off hand.

oliwer committed:
Test for strict equality before using a regex

This guaranties we get the right result in case a channel name
is the same as another channel prefix. Also limit the number
of results to 1.

oliwer commented:
You're right. There is little chance this may ever happen, but I modified the condition to make sure it never happens!
